### DISCUSSION TITLE
Using pycord as Discord API client as opposed to bot framework

### DISCUSSION THREAD

**Discussion by honzajavorek** [Pycord-Development/pycord#2382] [Q&A] [answered]
If I was to use pycord just as an API client as opposed to bot framework, what would be some good recommendations? How can I optimize it? I want to use it in simple one-time scripts (async/await is okay). I need no commands, events, websockets, endless loop. It's gonna start, do something, end. What would be the best way to setup pycord for such scenario?

**Comment by NeloBlivion** [comment, marked as answer]
Our most simple construct is `discord.Client`, but that still recieves events; Pycord isn't really built to support standalone requests. Your best bet would be to look through [http.py](https://github.com/Pycord-Development/pycord/blob/master/discord/http.py) and try authenticate it to run methods without going through Client, but we can't guarantee anything.

**Reply by honzajavorek** [reply]
Thanks. I'd probably avoid going through a classic HTTP client, the amount of heavy lifting pycord does is still a big help. I'll explore how to use `Client` more independently. If it still receives events, perhaps I can limit the amount of them somehow? Intents or something? 🤔 Or otherwise optimize connecting?

**Reply by NeloBlivion** [reply]
Indeed, you can pass `Intents.none()` to restrict the events; but you will still get *something*, the websocket connection will still remain open etc. The `http.py` I linked to is from the lib, contains (almost?) every single method that the library uses over the API; that's probably what you want (recall that `Client.start` typically only takes the token for authentication, so figuring it out yourself shouldn't be too hard)

**Reply by honzajavorek** [reply]
Thanks, that's good pointers.
